pkg/report/sarif.go

```diff
@@ -218,6 +214,37 @@ func (sw SarifWriter) Write(report types.Report) error {
 	return sarifReport.PrettyWrite(sw.Output)
 }
 
+func toSarifLocations(locations []location, artifactLocation, locationMessage string) []*sarif.Location {
```

Comment:
there is a lot of duplicated code. isn't it? I think we have to refactor this block. let's discuss it.

Reply:
Removed duplicated code. Take a look please.